### ISSUE TITLE
`Style/ExactRegexpMatch` registers false positive offenses

### ISSUE THREAD

**Issue by RDeckard** [rubocop/rubocop#11880]
## Expected behavior

`Style/ExactRegexpMatch` should not register false positive offenses.

## Actual behavior

`Style/ExactRegexpMatch` registers an offense with e.g. this regex: `/\A0+\z/`.

It makes the auto-correction changes e.g.:
```
"00".match?(/\A0+\z/) # => true
```
to
```
"00" == '0' # => false
```
which is wrong. 🙂

## Steps to reproduce the problem

Adding this example in `spec/rubocop/cop/style/exact_regexp_match_spec.rb`:
``` ruby
  it 'does not register an offense when using `string =~ /\A0+\z/`' do
    expect_no_offenses(<<~'RUBY')
      string =~ /\A0+\z/
    RUBY
  end
```

Make the specs fail:
<img width="1135" alt="image" src="https://github.com/rubocop/rubocop/assets/5458964/c58e4aab-309c-46b2-b2da-afdbd029206b">

Sorry to not be able to go further and fix this myself, but I'm not so used to the `Rubocop` parser and I actually don't understand why the guard clause `tokens[0] == :bos && tokens[1] == :literal && tokens[2] == :eos` doesn't work properly. 🤷‍♂️

## RuboCop version

```
$ bundle exec rubocop -V
1.51.0 (using Parser 3.2.2.1, rubocop-ast 1.28.1, running on ruby 3.2.1) [arm64-darwin21]
  - rubocop-performance 1.17.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.22.0
```
